exporters/ostream/test/ostream_log_test.cc

```diff
@@ -0,0 +1,263 @@
+{
+namespace logs
+{
+
+// Test that processor/exporter is shutdown, no logs should be sent to stream
+TEST(OStreamLogExporter, Shutdown)
+{
+  auto exporter =
+      std::unique_ptr<sdklogs::LogExporter>(new opentelemetry::exporter::logs::OStreamLogExporter);
+  auto processor =
+      std::shared_ptr<sdklogs::LogProcessor>(new sdklogs::SimpleLogProcessor(std::move(exporter)));
+
+  auto record  = std::shared_ptr<logs_api::LogRecord>(new logs_api::LogRecord());
+  record->name = "Test Log";
+
+  // Create stringstream to redirect to
+  std::stringstream stdoutOutput;
+
+  // Save cout's buffer here
+  std::streambuf *sbuf = std::cout.rdbuf();
+
+  // Redirect cout to our stringstream buffer
+  std::cout.rdbuf(stdoutOutput.rdbuf());
+
+  processor->Shutdown();
+
+  // After processor/exporter is shutdown, no logs should be sent to stream
+  processor->OnReceive(record);
+
+  std::cout.rdbuf(sbuf);
+
+  ASSERT_EQ(stdoutOutput.str(), "");
+}
+
+// ---------------------------------- Print to cout, cerr, and clog -------------------------
+
+// Print Log to std::cout
+TEST(OStreamLogExporter, PrintSimpleLogToCout)
+{
+  // Initialize an Ostream exporter to cout
+  auto exporter = std::unique_ptr<sdklogs::LogExporter>(
+      new opentelemetry::exporter::logs::OStreamLogExporter(std::cout));
+  auto processor =
+      std::shared_ptr<sdklogs::LogProcessor>(new sdklogs::SimpleLogProcessor(std::move(exporter)));
+
+  // Save original stream buffer and redirect cout to our new stream buffer
+  std::streambuf *sbuf = std::cout.rdbuf();
+  std::stringstream stdcoutOutput;
+  std::cout.rdbuf(stdcoutOutput.rdbuf());
+
+  // Create a log record and manually set all fields (since we are not using SDK to inject fields)
+  opentelemetry::core::SystemTimestamp now(std::chrono::system_clock::now());
+
+  auto record       = std::shared_ptr<logs_api::LogRecord>(new logs_api::LogRecord());
+  record->timestamp = now;
+  record->severity  = logs_api::Severity::kInfo;
+  record->name      = "Test Log";
+  record->body      = "Message";
+
+  // Log a record to cout
+  processor->OnReceive(record);
+
+  // Reset cout's original stringstream buffer
+  std::cout.rdbuf(sbuf);
+
+  std::string expectedOutput =
+      "{\n"
+      "    timestamp   : " +
+      std::to_string(now.time_since_epoch().count()) +
+      "\n"
+      "    severity    : kInfo\n"
+      "    name        : Test Log\n"
+      "    body        : Message\n"
+      "    resource    : {}\n"
+      "    attributes  : {}\n"
+      "    trace_id    : 00000000000000000000000000000000\n"
+      "    span_id     : 0000000000000000\n"
+      "    trace_flags : 00\n"
+      "}\n";
+  ASSERT_EQ(stdcoutOutput.str(), expectedOutput);
+}
+
+// Print log to std::cerr
+TEST(OStreamLogExporter, PrintLogWithResourceToCerr)
+{
+  // Initialize an Ostream exporter to cerr
+  auto exporter = std::unique_ptr<sdklogs::LogExporter>(
+      new opentelemetry::exporter::logs::OStreamLogExporter(std::cerr));
+  auto processor =
```

Comment:
nit: call the exporter's Export() directly to avoid having to create a processor instance?

Reply:
Thanks, great suggestion! I'll leave testing with a processor for the Integration Test below